auth-tokens-filter/src/main/java/com/palantir/tokens/auth/http/BearerTokenLoggingContextFilter.java

```diff
@@ -0,0 +1,97 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.tokens.auth.http;
+
+import com.google.common.net.HttpHeaders;
+import com.palantir.tokens.auth.AuthHeader;
+import com.palantir.tokens.auth.UnverifiedJsonWebToken;
+import java.io.IOException;
+import java.security.Principal;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import org.eclipse.jetty.security.DefaultUserIdentity;
+import org.eclipse.jetty.security.UserAuthentication;
+import org.eclipse.jetty.server.Request;
+import org.eclipse.jetty.server.UserIdentity;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.slf4j.MDC;
+
+/**
+ * A {@link Filter} that inserts user information into slf4j logging context and standard HTTP request logging context.
+ */
+public final class BearerTokenLoggingContextFilter implements Filter {
+
+    private static final Logger log = LoggerFactory.getLogger(BearerTokenLoggingContextFilter.class);
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {}
+
+    @Override
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
+            throws IOException, ServletException {
+        if (request instanceof HttpServletRequest) {
+            HttpServletRequest httpRequest = (HttpServletRequest) request;
+            String rawAuthHeader = httpRequest.getHeader(HttpHeaders.AUTHORIZATION);
+            if (rawAuthHeader != null) {
+                try {
+                    UnverifiedJsonWebToken jwt = UnverifiedJsonWebToken.of(
+                            AuthHeader.valueOf(rawAuthHeader).getBearerToken());
+
+                    // slf4j
+                    MDC.put("userId", jwt.getUnverifiedUserId());
+
+                    // Jetty
+                    if (request instanceof org.eclipse.jetty.server.Request) {
+                        Request jettyRequest = (org.eclipse.jetty.server.Request) request;
+
+                        UserIdentity userIdentity = new DefaultUserIdentity(null,
+                                new UsernamePrincipal(jwt.getUnverifiedUserId()), new String[0]);
+
+                        jettyRequest.setAuthentication(new UserAuthentication("BEARER-TOKEN", userIdentity));
+                    }
+
+                } catch (Throwable t) {
+                    log.error("Unable to decode authorization header", t);
```

Comment:
OK, if we hold ourselves to
https://github.com/palantir-baseline/best-practices/blob/master/java-coding-guidelines/readme.md#log-levels
,
then either error or warn would be fine, I guess.

On Fri, Jun 10, 2016 at 2:47 PM Mark Elliot <notifications@github.com>
wrote:

> In
> auth-tokens-filter/src/main/java/com/palantir/tokens/auth/http/BearerTokenLoggingContextFilter.java
> <https://github.com/palantir/auth-tokens/pull/11#discussion_r66606927>:
>
> > +
> > +                    // slf4j
> > +                    MDC.put("userId", jwt.getUnverifiedUserId());
> > +
> > +                    // Jetty
> > +                    if (request instanceof org.eclipse.jetty.server.Request) {
> > +                        Request jettyRequest = (org.eclipse.jetty.server.Request) request;
> > +
> > +                        UserIdentity userIdentity = new DefaultUserIdentity(null,
> > +                                new UsernamePrincipal(jwt.getUnverifiedUserId()), new String[0]);
> > +
> > +                        jettyRequest.setAuthentication(new UserAuthentication("BEARER-
>
> TOKEN", userIdentity));
> > +                    }
> > +
> > +                } catch (Throwable t) {
> > +                    log.error("Unable to decode authorization header", t);
>
> If you get here it means we weren't able to decode a provided bearer
> token. That seems worth investigating?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/palantir/auth-tokens/pull/11/files/b9eae140e2106654faeb97ba06c0911bf7cde963#r66606927>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AGOdwbprYKjDIBzQDEN4LVseb6GYaucTks5qKVzAgaJpZM4IyizV>
> .
>